src/FlatSharp/TypeModel/Vectors/UnityNativeArrayVectorTypeModel.cs

```diff
@@ -0,0 +1,57 @@
+namespace FlatSharp.TypeModel.Vectors;
+
+/// <summary>
+/// Defines a vector type model for a Unity NativeArray collection.
+/// </summary>
+public class UnityNativeArrayVectorTypeModel : BaseVectorTypeModel
+{
+    internal UnityNativeArrayVectorTypeModel(Type vectorType, TypeModelContainer provider) : base(vectorType, provider)
+    {
+    }
+
+    public override string LengthPropertyName => nameof(Memory<byte>.Length);
+
+    public override Type OnInitialize()
+    {
+        return this.ClrType.GetGenericArguments()[0];
+    }
+
+    protected override string CreateLoop(FlatBufferSerializerOptions options, string vectorVariableName, string numberOfItemsVariableName, string expectedVariableName, string body)
+    {
+        FlatSharpInternal.Assert(false, "Not expecting to do loop get max size for memory vector");
+        throw new Exception();
+    }
+
+    public override CodeGeneratedMethod CreateParseMethodBody(ParserCodeGenContext context)
+    {
+        ValidateWriteThrough(
+            writeThroughSupported: false,
+            this,
+            context.AllFieldContexts,
+            context.Options);
+
+        var lazySuffix = context.Options.Lazy ? "Lazy" : "";
```

Comment:
Sorry -- it looks like you got caught up in a merge conflict here. FlatSharp 7 is going to have a different characterstic, which is to generate all serializers for all types. The broader goal is to reduce code duplication for better cache hits and branch predictor utilization for cases when two parent tables both reference the same inner table. In that case, FlatSharp 6 will generate an entirely different serializer "graph" for each parent table, so all shared dependencies have duplicate code.